impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultModelXmlFactory.java

```diff
@@ -121,7 +121,7 @@ public void write(XmlWriterRequest<Model> request) throws XmlWriterException {
         Writer writer = request.getWriter();
         Function<Object, String> inputLocationFormatter = request.getInputLocationFormatter();
         if (writer == null && outputStream == null && path == null) {
-            throw new IllegalArgumentException("writer, outputStream or path must be non null");
+            throw new IllegalArgumentException("writer, output stream, or path must be non-null");
```

Comment:
```suggestion
            throw new IllegalArgumentException("writer, output stream, or path must be non-null.");
```
missing `dot.`
<img width="1092" alt="image" src="https://github.com/user-attachments/assets/9cf2ebb3-76cb-4753-aa24-d3c1c93e1592" />